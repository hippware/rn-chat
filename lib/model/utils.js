"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const mobx_1 = require("mobx");
function waitFor(condition) {
    return tslib_1.__awaiter(this, void 0, void 0, function* () {
        return new Promise((resolve, reject) => {
            mobx_1.when(() => {
                let res = false;
                try {
                    res = condition();
                }
                catch (e) {
                    reject(e);
                }
                return res;
            }, () => {
                resolve();
            });
        });
    });
}
exports.waitFor = waitFor;
//# sourceMappingURL=utils.js.map