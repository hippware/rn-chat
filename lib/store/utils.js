"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const mobx_1 = require("mobx");
function pad(n, width, z = '0') {
    const str = n.toString();
    return str.length >= width ? str : new Array(width - str.length + 1).join(z) + n;
}
function process(result) {
    if (typeof result === 'object') {
        if (Array.isArray(result)) {
            return result.map(el => process(el));
        }
        else {
            if (result['#text'] && Object.keys(result).length === 1) {
                return result['#text'];
            }
            const res = {};
            let changed = false;
            // eslint-disable-next-line
            for (const key in result) {
                // eslint-disable-next-line
                if (result.hasOwnProperty(key)) {
                    changed = true;
                    res[key] = process(result[key]);
                }
            }
            if (!changed) {
                return undefined;
            }
            return res;
        }
    }
    else {
        return result;
    }
}
function isArray(res) {
    return Array.isArray(res) || mobx_1.isObservableArray(res);
}
exports.isArray = isArray;
function camelize(str) {
    return str
        .replace(/\W|_|\d/g, ' ')
        .replace(/(?:^\w|[A-Z]|\b\w)/g, (letter, index) => {
        return index === 0 ? letter.toLowerCase() : letter.toUpperCase();
    })
        .replace(/\s+/g, '');
}
exports.camelize = camelize;
function processMap(data) {
    const res = {};
    Object.keys(data).forEach(key => {
        const value = data[key];
        try {
            if (value && value !== 'null' && key !== 'field') {
                if (key === 'roles') {
                    res.roles = isArray(data.roles.role) ? data.roles.role : [data.roles.role];
                }
                else if (['followers', 'bots', 'followed'].indexOf(key) !== -1) {
                    res[key + 'Size'] = parseInt(data[key].size);
                }
                else if (data[key].thumbnail_url !== undefined) {
                    // we have image here!
                    if (data[key]['#text']) {
                        res[key] = { id: data[key]['#text'], url: data[key].thumbnail_url };
                    }
                }
                else if (key === 'subscribed') {
                    res.isSubscribed = value === 'true';
                }
                else if (key === 'owner') {
                    res.owner = Strophe.getNodeFromJid(value);
                }
                else if (key === 'subscribers') {
                    res.followersSize = parseInt(value.size);
                }
                else if (key === 'location') {
                    res.location = { latitude: parseFloat(value.geoloc.lat), longitude: parseFloat(value.geoloc.lon) };
                }
                else if (key === 'updated') {
                    res.time = iso8601toDate(value).getTime();
                }
                else if (key === 'radius') {
                    res.radius = parseFloat(value);
                }
                else {
                    const numbers = ['image_items', 'total_items', 'visibility'];
                    res[camelize(key)] = numbers.indexOf(key) !== -1 ? parseInt(value) : value;
                }
            }
        }
        catch (e) {
            console.error(`Cannot process key ${key} value: ${value}`);
        }
    });
    delete res.id;
    return res;
}
exports.processMap = processMap;
function fromCamelCase(data = {}) {
    const { firstName, userID, phoneNumber, lastName, sessionID, uuid } = data, result = tslib_1.__rest(data, ["firstName", "userID", "phoneNumber", "lastName", "sessionID", "uuid"]);
    if (phoneNumber) {
        result.phone_number = phoneNumber;
        result.phoneNumber = phoneNumber;
    }
    if (userID) {
        result.auth_user = userID;
    }
    if (firstName) {
        result.first_name = firstName;
    }
    if (lastName) {
        result.last_name = lastName;
    }
    if (sessionID) {
        result.token = sessionID;
    }
    if (uuid) {
        result.user = uuid;
    }
    return result;
}
exports.fromCamelCase = fromCamelCase;
function getJid(username, host, resource) {
    let jid = `${username}@${host}`;
    if (resource) {
        jid = `${jid}/${resource}`;
    }
    return jid;
}
exports.getJid = getJid;
function getNodeJid(jid) {
    if (jid.indexOf('@') < 0) {
        return jid;
    }
    return jid.split('@')[0];
}
exports.getNodeJid = getNodeJid;
/** Function: getUniqueId
 *  Generate a unique ID for use in <iq/> elements.
 *
 *  All <iq/> stanzas are required to have unique id attributes.  This
 *  function makes creating these easy.  Each connection instance has
 *  a counter which starts from zero, and the value of this counter
 *  plus a colon followed by the suffix becomes the unique id. If no
 *  suffix is supplied, the counter is used as the unique id.
 *
 *  Suffixes are used to make debugging easier when reading the stream
 *  data, and their use is recommended.  The counter resets to 0 for
 *  every new connection for the same reason.  For connections to the
 *  same server that authenticate the same way, all the ids should be
 *  the same, which makes it easy to see changes.  This is useful for
 *  automated testing as well.
 *
 *  Parameters:
 *    (String) suffix - A optional suffix to append to the id.
 *
 *  Returns:
 *    A unique string to be used for the id attribute.
 */
function getUniqueId(suffix) {
    let uuid = 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, c => {
        let r = (Math.random() * 16) | 0, // eslint-disable-line
        v = c === 'x' ? r : (r & 0x3) | 0x8; // eslint-disable-line
        return v.toString(16);
    });
    if (typeof suffix === 'string' || typeof suffix === 'number') {
        return `${uuid}:${suffix}`;
    }
    else {
        return `${uuid}`;
    }
}
exports.getUniqueId = getUniqueId;
function generateID() {
    const time = Date.now();
    return `s${time}${pad(Math.round(Math.random() * 1000), 4)}`;
}
exports.generateID = generateID;
function parseXml(xml, arrayTags) {
    function isArray(o) {
        return Object.prototype.toString.apply(o) === '[object Array]';
    }
    function parseNode(xmlNode, res) {
        if (!xmlNode) {
            return;
        }
        if (xmlNode.nodeName === '#text') {
            /* if you want the object to have a properyty "#text" even if it is "",
                               remove that if-else and use code that is currently in else block
                               */
            if (xmlNode.nodeValue.trim() === '') {
                return;
            }
            else {
                res[xmlNode.nodeName] = xmlNode.nodeValue;
                return;
            }
        }
        const jsonNode = {};
        const existing = res[xmlNode.nodeName];
        if (existing) {
            if (!isArray(existing)) {
                res[xmlNode.nodeName] = [existing, jsonNode];
            }
            else {
                res[xmlNode.nodeName].push(jsonNode);
            }
        }
        else if (arrayTags && arrayTags.indexOf(xmlNode.nodeName) !== -1) {
            res[xmlNode.nodeName] = [jsonNode];
        }
        else {
            res[xmlNode.nodeName] = jsonNode;
        }
        if (xmlNode.attributes) {
            const len = xmlNode.attributes.length;
            for (let i = 0; i < len; i++) {
                const attribute = xmlNode.attributes[i];
                jsonNode[attribute.nodeName] = attribute.nodeValue;
            }
        }
        const length = xmlNode.childNodes.length;
        for (let i = 0; i < length; i++) {
            parseNode(xmlNode.childNodes[i], jsonNode);
        }
    }
    const result = {};
    parseNode(xml, result);
    return process(result);
}
exports.parseXml = parseXml;
// return hashcode for given string
function hashCode(s) {
    let hash = 0, i, chr, len;
    if (s.length === 0)
        return hash;
    for (i = 0, len = s.length; i < len; i++) {
        chr = s.charCodeAt(i);
        hash = (hash << 5) - hash + chr; // eslint-disable-line
        // Convert to 32bit integer.
        hash = Math.abs(hash & hash); // eslint-disable-line
    }
    return hash;
}
exports.hashCode = hashCode;
function iso8601toDate(date) {
    let timestamp = Date.parse(date), minutesOffset = 0;
    if (isNaN(timestamp)) {
        // eslint-disable-next-line
        let struct = /^(\d{4}|[+\-]\d{6})-(\d{2})-(\d{2})(?:[T ](\d{2}):(\d{2})(?::(\d{2})(?:\.(\d{3,}))?)?(?:(Z)|([+\-])(\d{2})(?::?(\d{2}))?))?/.exec(date);
        if (struct) {
            if (struct[8] !== 'Z') {
                minutesOffset = +struct[10] * 60 + +struct[11];
                if (struct[9] === '+') {
                    minutesOffset = -minutesOffset;
                }
            }
            return new Date(+struct[1], +struct[2] - 1, +struct[3], +struct[4], +struct[5] + minutesOffset, +struct[6], struct[7] ? +struct[7].substr(0, 3) : 0);
        }
        else {
            // XEP-0091 dateAsString
            timestamp = Date.parse(`${date.replace(/^(\d{4})(\d{2})(\d{2})/, '$1-$2-$3')}Z`);
        }
    }
    return new Date(timestamp);
}
exports.iso8601toDate = iso8601toDate;
//# sourceMappingURL=utils.js.map