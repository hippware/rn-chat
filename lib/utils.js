"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
function process(result) {
    if (typeof result === 'object') {
        if (Array.isArray(result)) {
            return result.map(function (el) { return process(el); });
        }
        else {
            if (result['#text'] && Object.keys(result).length === 1) {
                return result['#text'];
            }
            var res = {};
            var changed = false;
            // eslint-disable-next-line
            for (var key in result) {
                // eslint-disable-next-line
                if (result.hasOwnProperty(key)) {
                    changed = true;
                    res[key] = process(result[key]);
                }
            }
            if (!changed) {
                return undefined;
            }
            return res;
        }
    }
    else {
        return result;
    }
}
exports.default = {
    fromCamelCase: function (data) {
        if (data === void 0) { data = {}; }
        var firstName = data.firstName, userID = data.userID, phoneNumber = data.phoneNumber, lastName = data.lastName, sessionID = data.sessionID, uuid = data.uuid, result = tslib_1.__rest(data, ["firstName", "userID", "phoneNumber", "lastName", "sessionID", "uuid"]);
        if (phoneNumber) {
            result.phone_number = phoneNumber;
            result.phoneNumber = phoneNumber;
        }
        if (userID) {
            result.auth_user = userID;
        }
        if (firstName) {
            result.first_name = firstName;
        }
        if (lastName) {
            result.last_name = lastName;
        }
        if (sessionID) {
            result.token = sessionID;
        }
        if (uuid) {
            result.user = uuid;
        }
        return result;
    },
    getJid: function (username, host, resource) {
        var jid = username + "@" + host;
        if (resource) {
            jid = jid + "/" + resource;
        }
        return jid;
    },
    getNodeJid: function (jid) {
        if (jid.indexOf('@') < 0) {
            return null;
        }
        return jid.split('@')[0];
    },
    /** Function: getUniqueId
     *  Generate a unique ID for use in <iq/> elements.
     *
     *  All <iq/> stanzas are required to have unique id attributes.  This
     *  function makes creating these easy.  Each connection instance has
     *  a counter which starts from zero, and the value of this counter
     *  plus a colon followed by the suffix becomes the unique id. If no
     *  suffix is supplied, the counter is used as the unique id.
     *
     *  Suffixes are used to make debugging easier when reading the stream
     *  data, and their use is recommended.  The counter resets to 0 for
     *  every new connection for the same reason.  For connections to the
     *  same server that authenticate the same way, all the ids should be
     *  the same, which makes it easy to see changes.  This is useful for
     *  automated testing as well.
     *
     *  Parameters:
     *    (String) suffix - A optional suffix to append to the id.
     *
     *  Returns:
     *    A unique string to be used for the id attribute.
     */
    getUniqueId: function (suffix) {
        var uuid = 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function (c) {
            var r = (Math.random() * 16) | 0, // eslint-disable-line
            v = c === 'x' ? r : (r & 0x3) | 0x8; // eslint-disable-line
            return v.toString(16);
        });
        if (typeof suffix === 'string' || typeof suffix === 'number') {
            return uuid + ":" + suffix;
        }
        else {
            return "" + uuid;
        }
    },
    parseXml: function (xml, arrayTags) {
        function isArray(o) {
            return Object.prototype.toString.apply(o) === '[object Array]';
        }
        function parseNode(xmlNode, res) {
            if (!xmlNode) {
                return;
            }
            if (xmlNode.nodeName === '#text') {
                /* if you want the object to have a properyty "#text" even if it is "",
                                 remove that if-else and use code that is currently in else block
                                 */
                if (xmlNode.nodeValue.trim() === '') {
                    return;
                }
                else {
                    res[xmlNode.nodeName] = xmlNode.nodeValue;
                    return;
                }
            }
            var jsonNode = {};
            var existing = res[xmlNode.nodeName];
            if (existing) {
                if (!isArray(existing)) {
                    res[xmlNode.nodeName] = [existing, jsonNode];
                }
                else {
                    res[xmlNode.nodeName].push(jsonNode);
                }
            }
            else if (arrayTags && arrayTags.indexOf(xmlNode.nodeName) !== -1) {
                res[xmlNode.nodeName] = [jsonNode];
            }
            else {
                res[xmlNode.nodeName] = jsonNode;
            }
            if (xmlNode.attributes) {
                var len = xmlNode.attributes.length;
                for (var i = 0; i < len; i++) {
                    var attribute = xmlNode.attributes[i];
                    jsonNode[attribute.nodeName] = attribute.nodeValue;
                }
            }
            var length = xmlNode.childNodes.length;
            for (var i = 0; i < length; i++) {
                parseNode(xmlNode.childNodes[i], jsonNode);
            }
        }
        var result = {};
        parseNode(xml, result);
        return process(result);
    },
    // return hashcode for given string
    hashCode: function (s) {
        var hash = 0, i, chr, len;
        if (s.length === 0)
            return hash;
        for (i = 0, len = s.length; i < len; i++) {
            chr = s.charCodeAt(i);
            hash = (hash << 5) - hash + chr; // eslint-disable-line
            // Convert to 32bit integer.
            hash = Math.abs(hash & hash); // eslint-disable-line
        }
        return hash;
    },
    iso8601toDate: function (date) {
        var timestamp = Date.parse(date), minutesOffset = 0;
        if (isNaN(timestamp)) {
            // eslint-disable-next-line
            var struct = /^(\d{4}|[+\-]\d{6})-(\d{2})-(\d{2})(?:[T ](\d{2}):(\d{2})(?::(\d{2})(?:\.(\d{3,}))?)?(?:(Z)|([+\-])(\d{2})(?::?(\d{2}))?))?/.exec(date);
            if (struct) {
                if (struct[8] !== 'Z') {
                    minutesOffset = +struct[10] * 60 + +struct[11];
                    if (struct[9] === '+') {
                        minutesOffset = -minutesOffset;
                    }
                }
                return new Date(+struct[1], +struct[2] - 1, +struct[3], +struct[4], +struct[5] + minutesOffset, +struct[6], struct[7] ? +struct[7].substr(0, 3) : 0);
            }
            else {
                // XEP-0091 dateAsString
                timestamp = Date.parse(date.replace(/^(\d{4})(\d{2})(\d{2})/, '$1-$2-$3') + "Z");
            }
        }
        return new Date(timestamp);
    }
};
//# sourceMappingURL=utils.js.map