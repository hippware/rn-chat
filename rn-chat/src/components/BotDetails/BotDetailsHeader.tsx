import React from 'react'
import {View, Animated, Alert, Image, TouchableOpacity, StyleSheet, Clipboard} from 'react-native'
import {observer, inject} from 'mobx-react/native'
import {k, width, height} from '../Global'
import {colors} from '../../constants'
import {getSnapshot, isAlive} from 'mobx-state-tree'
import BotButtons from './BotButtons'
import UserInfoRow from './UserInfoRow'
import {RText} from '../common'
import BotDetailsMap from '../map/BotDetailsMap'
import {Actions} from 'react-native-router-flux'
import {IBot} from 'wocky-client'

type Props = {
  bot: IBot
  notificationStore?: any // TODO proper type
  analytics?: any // TODO proper type
  scale: number
}

type State = {
  fadeAnim: any
}

const DOUBLE_PRESS_DELAY = 300

@inject('notificationStore', 'analytics')
@observer
class BotDetailsHeader extends React.Component<Props, State> {
  lastImagePress?: number
  userInfo: any

  constructor(props: Props) {
    super(props)
    this.state = {
      fadeAnim: new Animated.Value(0),
    }
  }

  handleImagePress = () => {
    const now = new Date().getTime()

    if (this.lastImagePress && now - this.lastImagePress < DOUBLE_PRESS_DELAY) {
      delete this.lastImagePress
      this.handleImageDoublePress()
    } else {
      this.lastImagePress = now
    }
  }

  handleImageDoublePress = () => {
    if (!this.props.bot.isSubscribed) {
      this.subscribe()
    }
  }

  unsubscribe = () => {
    Alert.alert('', 'Are you sure you want to remove this from your saved bots?', [
      {text: 'Cancel', style: 'cancel'},
      {
        text: 'Remove',
        style: 'destructive',
        onPress: () => this.props.bot.unsubscribe(),
      },
    ])
  }

  subscribe = () => {
    this.props.bot.subscribe()
    this.setState({fadeAnim: new Animated.Value(1)})
    setTimeout(() => {
      Animated.timing(this.state.fadeAnim, {toValue: 0}).start()
    }, 500)
    this.props.analytics.track('bot_save', getSnapshot(this.props.bot))
  }

  copyAddress = () => {
    Clipboard.setString(this.props.bot.address)
    this.props.notificationStore.flash('Address copied to clipboard üëç')
  }

  render() {
    const {bot, scale} = this.props
    if (!bot || !isAlive(bot)) return null
    return (
      <View style={{flex: 1}}>
        <View
          style={{
            height: scale === 0 ? height - 60 * k : width,
            backgroundColor: 'white',
            overflow: 'hidden',
          }}
        >
          <BotDetailsMap
            bot={bot}
            onMapPress={() => Actions.refresh({scale: 0})}
            onImagePress={() => {
              Actions.refresh({scale: scale === 1 ? 0.5 : scale + 0.5})
            }}
            scale={scale}
          />
          <Animated.View
            pointerEvents="none"
            style={[{opacity: this.state.fadeAnim}, styles.botAddedContainer]}
          >
            <Image source={require('../../../images/iconBotAdded.png')} />
          </Animated.View>
        </View>
        {scale > 0 && (
          <View>
            <View style={styles.rowContainer}>
              <BotButtons
                bot={bot}
                subscribe={this.subscribe}
                unsubscribe={this.unsubscribe}
                isSubscribed={bot.isSubscribed}
                copyAddress={this.copyAddress}
              />
              <UserInfoRow bot={bot} copyAddress={this.copyAddress} />
              {!!bot.description && (
                <View style={styles.descriptionContainer}>
                  <RText numberOfLines={0} size={16} weight="Light" color={colors.DARK_PURPLE}>
                    {bot.description}
                  </RText>
                </View>
              )}
              {bot.guest && <GeofenceCTA bot={bot} />}
            </View>
            <View
              style={[
                styles.rowContainer,
                {
                  height: 45,
                  alignSelf: 'stretch',
                  marginTop: 8.5,
                  flexDirection: 'row',
                  alignItems: 'center',
                },
              ]}
            >
              <Image
                style={{width: 14, height: 14}}
                source={require('../../../images/postsIcon.png')}
              />
              <RText
                size={15}
                color={colors.DARK_PURPLE}
                style={{marginLeft: 7, letterSpacing: 0.3}}
              >
                Posts
              </RText>
              <RText size={12} color={colors.DARK_GREY} style={{marginLeft: 7}}>
                {bot.totalItems}
              </RText>
            </View>
            <View style={{height: 1, width}} />
          </View>
        )}
      </View>
    )
  }
}

const GeofenceCTA = observer(({bot}) => (
  <View style={{flexDirection: 'row'}}>
    <Image source={require('../../../images/footOpaquePink.png')} style={{width: 26, height: 34}} />
    <TouchableOpacity onPress={() => Actions.visitors({item: bot.id})}>
      <View style={{marginLeft: 10 * k}}>
        <RText color={colors.PINK} size={15}>
          See Who's Here
        </RText>
        <RText color={colors.DARK_GREY} size={12}>
          {bot.visitorsSize} people
        </RText>
      </View>
    </TouchableOpacity>
  </View>
))

export default BotDetailsHeader

const styles = StyleSheet.create({
  rowContainer: {
    backgroundColor: 'white',
    padding: 20 * k,
  },
  descriptionContainer: {
    paddingBottom: 15 * k,
  },
  botAddedContainer: {
    height: width,
    position: 'absolute',
    left: 0,
    right: 0,
    justifyContent: 'center',
    alignItems: 'center',
  },
})
