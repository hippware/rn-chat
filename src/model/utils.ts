import {when} from 'mobx'

export async function waitFor(condition: () => boolean) {
  return new Promise((resolve, reject) => {
    when(
      () => {
        let res = false
        try {
          res = condition()
        } catch (e) {
          reject(e)
        }
        return res
      },
      () => {
        resolve()
      }
    )
  })
}
