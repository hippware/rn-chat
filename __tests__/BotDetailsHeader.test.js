// @flow
/* global test */

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'mobx-react/native';
import {Bot} from 'wocky-client';
import './utils/mockMap';
import injects from './utils/inject-mocks';
import mockLocationStore from './utils/mockLocationStore';
import wocky from './utils/mockWocky';

import BotDetailsHeader from '../src/components/BotDetails/BotDetailsHeader';

describe('BotDetailsHeader', () => {
  test('renders with test bot', async () => {
    await wocky.login('user', 'password', 'host');
    const bot = Bot.create({
      id: '1234',
      isSubscribed: false,
      title: 'title',
      server: 'server',
      radius: 30,
      owner: null,
      // image: FileRef,
      // description: types.maybe(types.string),
      // visibility: VISIBILITY_PUBLIC,
      // location: types.maybe(Location),
      // address: '',
      // followersSize: 0,
      // totalItems: 0,
      // addressData: types.optional(Address, {}),
      // subscribers: types.optional(ProfilePaginableList, {}),
      // posts: types.optional(BotPostPaginableList, {}),
      // error: ''
    });
    mockLocationStore.setPosition({
      coords: {longitude: 2, latitude: 2, accuracy: 2},
    });
    const toRender = (
      <Provider wocky={wocky} {...injects} locationStore={mockLocationStore}>
        <BotDetailsHeader bot={bot} scale={0.5} />
      </Provider>
    );
    const tree = renderer.create(toRender).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
