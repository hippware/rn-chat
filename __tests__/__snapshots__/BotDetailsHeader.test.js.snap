// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BotDetailsHeader renders with test bot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "height": 750,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "bottom": -420,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <MapView
        autoZoom={true}
        bot={
          Object {
            "address": "",
            "addressData": Object {
              "city": "",
              "country": "",
              "county": "",
              "state": "",
            },
            "description": null,
            "error": "",
            "followersSize": 0,
            "geofence": false,
            "id": "1234",
            "image": null,
            "isSubscribed": false,
            "location": null,
            "owner": null,
            "posts": Object {
              "count": null,
              "result": Array [],
            },
            "radius": 30,
            "server": "server",
            "subscribers": Object {
              "count": null,
              "result": Array [],
            },
            "title": "title",
            "totalItems": 0,
            "visibility": 100,
          }
        }
        fullMap={false}
        initialRegion={
          Object {
            "latitude": 2,
            "latitudeDelta": 0.2,
            "longitude": 2,
            "longitudeDelta": 0.2,
          }
        }
        location={null}
        locationStore={Object {}}
        marker={
          <_class
            bot={
              Object {
                "address": "",
                "addressData": Object {
                  "city": "",
                  "country": "",
                  "county": "",
                  "state": "",
                },
                "description": null,
                "error": "",
                "followersSize": 0,
                "geofence": false,
                "id": "1234",
                "image": null,
                "isSubscribed": false,
                "location": null,
                "owner": null,
                "posts": Object {
                  "count": null,
                  "result": Array [],
                },
                "radius": 30,
                "server": "server",
                "subscribers": Object {
                  "count": null,
                  "result": Array [],
                },
                "title": "title",
                "totalItems": 0,
                "visibility": 100,
              }
            }
            onImagePress={[Function]}
            onMapPress={[Function]}
            scale={0.5}
          />
        }
        onImagePress={[Function]}
        onMapPress={[Function]}
        onMarkerSelect={[Function]}
        onPress={[Function]}
        onRegionChangeComplete={[Function]}
        pitchEnabled={false}
        rotateEnabled={false}
        scale={0.5}
        scrollEnabled={false}
        showOnlyBot={true}
        showUser={false}
        style={
          Object {
            "flex": 1,
          }
        }
        wocky={
          Object {
            "bots": Object {
              "storage": Object {},
            },
            "chats": Object {
              "_list": Array [],
            },
            "events": Object {
              "count": null,
              "result": Array [],
            },
            "files": Object {
              "storage": Object {},
            },
            "geoBots": Object {},
            "host": "host",
            "id": "wocky",
            "password": "password",
            "profile": Object {
              "avatar": null,
              "botsSize": 0,
              "email": "",
              "firstName": "",
              "followedSize": 0,
              "followersSize": 0,
              "handle": "",
              "id": "user",
              "isBlocked": false,
              "isFollowed": false,
              "isFollower": false,
              "isNew": false,
              "lastName": "",
              "loaded": true,
              "phoneNumber": "",
              "roles": Array [],
              "status": "available",
            },
            "profiles": Object {
              "storage": Object {
                "user": Object {
                  "avatar": null,
                  "botsSize": 0,
                  "firstName": "",
                  "followedSize": 0,
                  "followersSize": 0,
                  "handle": "",
                  "id": "user",
                  "isBlocked": false,
                  "isFollowed": false,
                  "isFollower": false,
                  "isNew": false,
                  "lastName": "",
                  "loaded": false,
                  "roles": Array [],
                  "status": "unavailable",
                },
              },
            },
            "roster": Object {},
            "sessionCount": 1,
            "updates": Array [],
            "username": "user",
            "version": "",
          }
        }
        zoomEnabled={false}
      />
      <View
        collapsable={undefined}
        style={
          Object {
            "backgroundColor": "#007bff",
            "borderBottomWidth": 1,
            "borderColor": "#006acd",
            "bottom": null,
            "left": 0,
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "translateY": -1334,
              },
            ],
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 1,
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "flex": 1,
                "flexDirection": "row",
                "padding": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "alignSelf": "stretch",
                  "flex": 1,
                  "flexDirection": "column",
                  "justifyContent": "center",
                  "marginLeft": 10,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      collapsable={undefined}
      pointerEvents="none"
      style={
        Object {
          "alignItems": "center",
          "height": 750,
          "justifyContent": "center",
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../images/iconBotAdded.png",
          }
        }
      />
    </View>
  </View>
  <View>
    <View
      style={
        Object {
          "height": 8.5,
          "width": 750,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "flexDirection": "row",
          "height": 45,
          "width": 750,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../images/postsIcon.png",
          }
        }
        style={
          Object {
            "height": 14,
            "marginLeft": 14,
            "width": 14,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "rgb(63,50,77)",
              "fontFamily": "Roboto-Regular",
              "fontSize": 15,
            },
            Object {
              "letterSpacing": 0.3,
              "marginLeft": 7,
            },
          ]
        }
      >
        Posts
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "rgb(155,155,155)",
              "fontFamily": "Roboto-Regular",
              "fontSize": 12,
            },
            Object {
              "marginLeft": 7,
            },
          ]
        }
      >
        0
      </Text>
    </View>
    <View
      style={
        Object {
          "height": 1,
          "width": 750,
        }
      }
    />
  </View>
</View>
`;
